pkg/ddl/executor.go

```diff
@@ -886,34 +860,91 @@ func GetDefaultCollation(cs string, defaultUTF8MB4Collation string) (string, err
 }
 
 // ResolveCharsetCollation will resolve the charset and collate by the order of parameters:
-// * If any given ast.CharsetOpt is not empty, the resolved charset and collate will be returned.
-// * If all ast.CharsetOpts are empty, the default charset and collate will be returned.
+//
+// charsetOpts[0] == explicitly set charset/collation for the object (object can be a table, database, etc)
```

Comment:
I'm thinking of simplify `charsetOpts` to say the options of smaller index has higher priority, so we don't need to explain `[0]`, `[1]`

And the function can handle `charsetOpts` like

1 for each charsetOpts
1.1 if the result charset is not set, use current `charsetOpt` to set it. Also check `ErrCollationCharsetMismatch` for the cases that collation is set by `charsetOpts[0]` and we are setting charset by `charsetOpts[1]`
1.2 if the result collation is not set, use current `charsetOpt` to set it. Also check `ErrCollationCharsetMismatch` and set corresponding charset of this collation
2 if only the charset is set, choose default collation
3 if both are not set, choose default charset and collation